src/libNode/FinalBlockProcessing.cpp

```diff
@@ -692,6 +700,8 @@ bool Node::ProcessFinalBlockCore(const bytes& message, unsigned int offset,
 
   bool toSendTxnToLookup = false;
 
+  bool toSendPendingTxn = !(GetUnconfirmedTxns().empty());
```

Comment:
The construction of return object from `GetUnconfirmedTxns` can be quite expensive.  Maybe you can create a function that returns the size of `m_unconfirmedTxns`, since the check for empty is done in more than one place.